pkg/pquerier/Series.go

```diff
@@ -0,0 +1,103 @@
+package pquerier
+
+import (
+	"github.com/v3io/v3io-tsdb/pkg/aggregate"
+	"github.com/v3io/v3io-tsdb/pkg/utils"
+	"math"
+)
+
+func NewDataFrameColumnSeries(indexColumn, dataColumn, countColumn Column, labels utils.Labels, hash uint64) *DataFrameColumnSeries {
+	s := &DataFrameColumnSeries{dataColumn: dataColumn, indexColumn: indexColumn, CountColumn: countColumn, labels: labels, key: hash}
+	s.iter = &DataFrameColumnSeriesIterator{indexColumn: indexColumn, dataColumn: dataColumn, countColumn: countColumn, currentIndex: -1}
+	return s
+}
+
+// This series converts two columns into a series of time-value pairs
+type DataFrameColumnSeries struct {
```

Comment:
you can just have the `DataFrameColumnSeries` as a member instead of copying all it properties

Reply:
It basically exposes an iterator so once i iterate it once it's useless. so if i'll call it twice second time is empty result.
Plus, there is a new `DataFrameColumnSeries` per every data column and i only pass pointers to the relevant columns.